### PR TITLE
feat: make CursorPagination component accessible

### PR DESCRIPTION
**Pull request by migueloller**
The changes here are heavily inspired by shadcn/ui Pagination component.

See https://ui.shadcn.com/docs/components/pagination.

https://github.com/user-attachments/assets/0a9712e4-05a8-4d97-ad93-c3d48922b35d

### CODE REVIEW

File: apps/web/vibes/soul/primitives/cursor-pagination/index.tsx [hunk 1]
```diff
@@ -15,7 +17,10 @@ export interface CursorPaginationInfo {
 }
 
 interface Props {
-  info: CursorPaginationInfo | Promise<CursorPaginationInfo>;
+  label?: Streamable<string | null>;
+  info: Streamable<CursorPaginationInfo>;
+  previousLabel?: Streamable<string | null>;
+  nextLabel?: Streamable<string | null>;
```
Comment (jorgemoya) [inline apps/web/vibes/soul/primitives/cursor-pagination/index.tsx:23]: Do you have an example of why these label values are streamable?

Comment (migueloller) [reply apps/web/vibes/soul/primitives/cursor-pagination/index.tsx:23]: It's because these are translated and getting translations is asynchronous

Comment (migueloller) [reply apps/web/vibes/soul/primitives/cursor-pagination/index.tsx:23]: As I was implementing changes to the category pages in Catalyst I realized that labels need to be streamable for this reason

Comment (jorgemoya) [reply apps/web/vibes/soul/primitives/cursor-pagination/index.tsx:23]: So do we need to make all labels Streamable?

Comment (migueloller) [reply apps/web/vibes/soul/primitives/cursor-pagination/index.tsx:23]: Yeah, probably. I'm planning on doing this as-needed as I work on Catalyst and VIBES

**Review by jorgemoya — APPROVED** [review]

**Pull request merged**